### PR TITLE
add flux python command to run configured python

### PR DESCRIPTION
**Pull request by trws**
This adds a command, `flux python`, to invoke the interpreter flux was configured with under the flux environment as it is set by the flux command.  It's a convenience wrapper essentially, but given the number of pythons that sometimes end up laying around in one's path, it can be helpful to get directly to the right one with a fixed PYTHON_PATH.

This also adjusted the environment setup to fix running python against
the build tree for out-of-tree builds.

### CODE REVIEW

File: src/common/libflux/Makefile.am [hunk 1]
```diff
@@ -34,7 +34,7 @@ intree_conf_cppflags = \
 	-DINTREE_EXEC_PATH=\"$(abs_top_builddir)/src/cmd:$(abs_top_srcdir)/src/cmd:$(abs_top_builddir)/src/broker\" \
 	-DINTREE_LUA_PATH_ADD=\"$(abs_top_srcdir)/src/bindings/lua/?.lua\" \
 	-DINTREE_LUA_CPATH_ADD=\"$(abs_top_builddir)/src/bindings/lua/?.so\" \
-	-DINTREE_PYTHON_PATH=\"$(abs_top_builddir)/src/bindings/python\" \
+	-DINTREE_PYTHON_PATH=\"$(abs_top_srcdir)/src/bindings/python:$(abs_top_builddir)/src/bindings/python\" \
```
Comment (SteVwonder) [inline src/common/libflux/Makefile.am]: Just to confirm, the srcdir is first because of environment using push_front?

Comment (trws) [reply src/common/libflux/Makefile.am]: Right.  This causes it to come out in the opposite order in PYTHON_PATH.

**Comment by SteVwonder** [comment]
LGTM!

**Comment by garlick** [comment]
Maybe it would be good to add a quick test to `t/t0001-basic.t`?

**Comment by garlick** [comment]
Thanks!  Maybe #1772 should go in first since it doesn't look like this handles the case where python isn't configured?

**Comment by trws** [comment]
Crud, yes you're right.  Thanks for that catch @garlick!

**Comment by garlick** [comment]
This one's next I think?  Just needs a rebase.

**Comment by trws** [comment]
Rebased, assuming travis is happy it should be GTG.

**Pull request merged**